### ISSUE TITLE
OverflowMenu traps focus in React preventing user from tabbing

### ISSUE THREAD

**Issue by electrostaticfleece** [carbon-design-system/carbon#4659]
## Detailed description

> What version of the Carbon Design System are you using?

Carbon Components React v7

> What did you expect to happen?

When attempting to tab either forward or backward when focused on the overflow menu, I would expect to be able to tab to the next or previous focusable element in the DOM. I was not.  

> What happened instead?

I was locked into the overflow menu element and was prevented from tabbing either forward or backward.

> What WCAG 2.1 checkpoint does the issue violate?

https://www.w3.org/WAI/WCAG21/quickref/#no-keyboard-trap

## Steps to reproduce the issue

1. Visit http://react.carbondesignsystem.com/?path=/story/overflowmenu--basic

2. Tab to the overflow menu button.

3. Attempt to tab to the next focusable element. This should not work.

4. Attempt to tab to the previously focusable element. This should not work.

5. Enable voice over commands locally

6. Tab to the overflow menu item again

7. No alternative options about how to escape the overflow menu and tab to the next element are provided.
